### ISSUE TITLE
Problem about the efficient about the AdvGAN

### ISSUE THREAD

**Issue by Polly2014** [qiuweimin1332499/AdvGAN_cifar10#1]
Hi, I'm very intrested in your work, and recently I tried to validate the advGAN in Imagenet dataset, but the result show that the efficient is not good. Then I used your git codes to test in cifar10, the result as bellow:
```sh
CUDA Available:  True
Files already downloaded and verified
training dataset:
num_correct:  41207
accuracy of adv imgs in training set: 0.824140

Files already downloaded and verified
testing dataset:
num_correct:  7626
accuracy of adv imgs in testing set: 0.762600
```
I'ts only 70%+ accuracy, even the normal FGSM etc. gradient-based method perform better than it.
Here I want to seek advice from you, that what is the reason or how to improve the performance about advGAN on cifar10 or Imagenet dataset.thx

**Comment by SoEzreal** [comment]
> Hi, I'm very intrested in your work, and recently I tried to validate the advGAN in Imagenet dataset, but the result show that the efficient is not good. Then I used your git codes to test in cifar10, the result as bellow:
> 
> ```shell
> CUDA Available:  True
> Files already downloaded and verified
> training dataset:
> num_correct:  41207
> accuracy of adv imgs in training set: 0.824140
> 
> Files already downloaded and verified
> testing dataset:
> num_correct:  7626
> accuracy of adv imgs in testing set: 0.762600
> ```
> 
> I'ts only 70%+ accuracy, even the normal FGSM etc. gradient-based method perform better than it.
> Here I want to seek advice from you, that what is the reason or how to improve the performance about advGAN on cifar10 or Imagenet dataset.thx

I have recently discovered this problem. Although I spent a lot of time on the training data of CIFAR10, I found little change in the attack effect on the test data (EPOCH =25 or EPOCH =600).On the other hand, FGSM and PGD have high attack intensity and significantly save time. Have you found a good way to improve advGAN? I don't think advGAN's training on training data is as good as FGSM, which directly attacks the test data.

**Comment by KuanKuanQAQ** [comment]
> > Hi, I'm very intrested in your work, and recently I tried to validate the advGAN in Imagenet dataset, but the result show that the efficient is not good. Then I used your git codes to test in cifar10, the result as bellow:
> > ```shell
> > CUDA Available:  True
> > Files already downloaded and verified
> > training dataset:
> > num_correct:  41207
> > accuracy of adv imgs in training set: 0.824140
> > 
> > Files already downloaded and verified
> > testing dataset:
> > num_correct:  7626
> > accuracy of adv imgs in testing set: 0.762600
> > ```
> > 
> > 
> > I'ts only 70%+ accuracy, even the normal FGSM etc. gradient-based method perform better than it.
> > Here I want to seek advice from you, that what is the reason or how to improve the performance about advGAN on cifar10 or Imagenet dataset.thx
> 
> I have recently discovered this problem. Although I spent a lot of time on the training data of CIFAR10, I found little change in the attack effect on the test data (EPOCH =25 or EPOCH =600).On the other hand, FGSM and PGD have high attack intensity and significantly save time. Have you found a good way to improve advGAN? I don't think advGAN's training on training data is as good as FGSM, which directly attacks the test data.

yep, I totally agree with them. I am also training an advGAN model recently, but the effect is quite poor. Loss function, alpha, clip, optimizer are all adjustable places, but I don't know how to adjust.

**Comment by renhl717445** [comment]
> 
Hello, have you solved it at last? I am also training this now, but the value of L_G_fake is gradually increasing. Is this normal? I also adjusted the relevant parameters, but the effect is not improved
